dom0-updates/qfile-dom0-unpacker.c

```diff
@@ -90,7 +91,20 @@ int main(int argc, char ** argv)
 		files_limit = atoll(var);
 
 	set_size_limit(bytes_limit, files_limit);
-	if (argc > 3 && strcmp(argv[3],"-v")==0)
-		set_verbose(1);
+    for (i = 3; i < argc; i++) {
+        if (strcmp(argv[i], "-v") == 0) {
+            set_verbose(1);
+        } else if (strcmp(argv[i], "-w") == 0) {
+            if (i+1 < argc && argv[i+1][0] != '-') {
+                set_wait_for_space(atoi(argv[i+1]));
```

Comment:
`atoi` doesn’t detect errors, and has undefined behavior on overflow.  `strtoul` is better, but we might just want to check for `0` and `1` explicitly and reject other values.  This code also doesn’t handle `-w0` and `-w1`.

Reply:
> reject other values

No, this option _does_ accept other values, specifically the free space that should be left (in bytes).

>  This code also doesn’t handle `-w0` and `-w1`.

Why should it? It isn't how _this program_ expects arguments. It is an internal tool not supposed to be called manually.

Reply:
Sorry, I didn’t think of that.  I still consider `strtoul` a better choice, since it has well-defined behavior and its return type (`unsigned long`) is what `set_wait_for_space` expects.

Reply:
Yes, `strtoul` is a good idea.